### ISSUE TITLE
[breaking] remove proxy object in favor of request.context

### ISSUE THREAD

**Issue by blampe** [uber/tchannel-python#76]
All of our handlers currently get a "proxy" / "tchannel" / "channel" argument that is conceptually confusing and error-prone. It's purpose is to tie together zipkin tracing information, but it's too easy for users to not use it. It *should* be a totally reasonable pattern to write some wrappers around service calls that use the global tchannel, versus needing to plumb this proxy object everywhere.

Let's make the proxy argument optional. For backwards compatibility maybe we could infer its presence from the handler's signature?

```
@app.register(...)
def foo(request, response, proxy):
    pass

@app.register(...)
def foo(request, response):
    pass
``` 

In the latter case, we should hook up a [StackContext](http://tornado.readthedocs.org/en/latest/_modules/tornado/stack_context.html) to the handler that tracks the current trace id / span id such that subsequent tchannel calls automatically send that information downstream.

**Comment by abhinav** [comment]
I agree with wanting to remove it, but don't think function signature inspection is the way to go. Function signatures get munged if you're using decorators that don't use the `wrapt` or `decorator` libraries. A boolean flag when specifying the decorator (`inject=True`, or a different name) might make sense while we're in the process of deprecating it. And after that we can just remove it completely.

Also, for the sake of clarity, we may want to start referring to the `response` argument as `response_writer` or something. Makes it clear that that's what you use to send responses.

**Comment by blampe** [comment]
Yes nspection is very hacky and fragile :)

Just trying to think whether we can do this in a back-compat way.

**Comment by breerly** [comment]
@abhinav I think `response_writer` is a huge improvement.

**Comment by breerly** [comment]
This should be unblockeed!

**Comment by blampe** [comment]
Fixed in #92.
